### ISSUE TITLE
LabelList `dataKey` prop appears to have incorrect type definition

### ISSUE THREAD

**Issue by dylanpyle** [recharts/recharts#5245]
## Do you want to request a feature or report a bug?

Report a bug

## What is the current behavior?

When passing a function as `dataKey` prop to `<LabelList />`, at runtime the function receives as an argument the data point for which to generate a label, in the same format/type as whatever data was passed into the chart.

This does not match the type definition, which claims it receives something which extends `Data`, defined as:

```
interface Data {
  value?: number | string | Array<number | string>;
  payload?: any;
  parentViewBox?: ViewBox;
}
```
(https://github.com/recharts/recharts/blob/9936484bbd8bdc8b1b56efa30a94206e5b5d8ab1/src/component/LabelList.tsx#L13C1-L17C2)

So, code like the following fails to compile:

```ts
interface MyDataPoint {
  name: string;
  n: number;
}

const data: MyDataPoint[] = [
  { name: "Monday", n: 1000 },
  { name: "Tuesday", n: 12 },
  { name: "Wednesday", n: 2221 },
  { name: "Thursday", n: 30 },
];

const getLabel = (data: MyDataPoint) => {
  return data.n;
};

export default function App() {
  return (
    <BarChart
      data={data}
    >
      <Bar dataKey="n">
        <LabelList dataKey={getLabel} />
      </Bar>
    </BarChart>
  );
}
```

— with the error:

```
Type '(data: MyDataPoint) => number' is not assignable to type 'DataKey<Data> | undefined'.
  Type '(data: MyDataPoint) => number' is not assignable to type '(obj: Data) => any'.
    Types of parameters 'data' and 'obj' are incompatible.
      Type 'Data' is missing the following properties from type 'MyDataPoint': name, n
```

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).

- [Codesandbox (working fine at runtime)](https://codesandbox.io/p/sandbox/simple-line-chart-forked-9rdvw8?file=%2Fsrc%2FApp.tsx)
- [TypeScript playground (showing compilation type error)](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAIgHQHoDOMCeAbApmpAYzTQQG4AoUSWOAJVwEND4AzKCERKJl8q8NHgBvOACFGUAMIALSTAA04yUoAaAQQAewNEoCaWnUoAyjAEa5sxnfAC+cdp265Cc2KUpUAdjFxRWzLhwALKYACKMMIwAChDAPnDCFHBwXowguABccBhQ8QDmlCle2V4AriAWUJS2FBSEEF4YcAAmkYzZoRFRsfEwANoAunAAvHD9yYmp6VmIwY1tmAhKJXAAjAAMW3C2CpOiaRnZCAAqZfiLy6nZawBMO3spBzPHAOq4LV4XjEsr2bcAtYPfbTI6IE4yMpQNCXP5wADMG2Bg08DSa8HyuBgpgs2FGcAAFG0op1wu1ej4AJSjAB8iUmPBgUK8rXaSC8NTquE0NHgLVwATK2DYZS8LGAjTg6jAYAJ1KSKUZzMJkxSAB4JNI3DBVSk4AB3YAtGAyEbCACsW1qer1MlwwHyMhgZsRG2tNtZUTNxMY7ptIEk+XiZoVHpSMAgYGy5seYbgeUdMGyiNjYbwrCTcFuG1THrMEBgEZA0dzKVsfrgNN16s1nsYAGlcJgRggvAgq3G4GqcZZrM0fY3m8JMdjzJZ7CgOx61ShNVObWqNNo0HXBy3DrgEHBJ9WuwZl6um+utzuF7PJLJ5FPKZygA)

## What is the expected behavior?

This code compiles

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?

Using latest Recharts (2.13.3), TypeScript (5.6.3)

**Comment by ckifer** [comment]
Feel free to submit a PR if you'd like

**Comment by ckifer** [comment]
fixed in the 3.x branch in PR https://github.com/recharts/recharts/pull/5250

if you need this fixed in 2.x please make a PR

**Comment by dylanpyle** [comment]
@ckifer thank you!! For the use case I was working on when I hit this I ended up not using LabelList, so a fix in 3.x doesn't block me. But really appreciate the quick response and turnaround.

**Comment by ckifer** [comment]
Sounds good, appreciate the report!

**Comment by ckifer** [comment]
eh easy enough so fixed in master as well 
https://github.com/recharts/recharts/pull/5252

not sure when it'll be released though

**Comment by ckifer** [comment]
released in 2.14.0
